### ISSUE TITLE
RCS1163 special case false positive: don't trigger for constructors on ISerializable classes

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#281]
Given the following class:

```csharp
public class AClass : ISerializable
{
    protected AClass(SerializationInfo info, StreamingContext context /* unused, triggers RCS1163 */)
    {
        DeserializeFrom(info);
    }

    public void GetObjectData(SerializationInfo info, StreamingContext context /* unused, no RCS1163 */)
    {
        SerializeTo(info);
    }
}
```

The `protected` constructor with those exact arguments is required to satisfy the requirements of the `ISerializable` interface, so whether its arguments are used or not is irrelevant, thus it should not trigger RCS1163.
